lib/ruby_lsp/setup_bundler.rb

```diff
@@ -221,13 +236,13 @@ def run_bundle_install(bundle_gemfile = @gemfile)
 
     sig { returns(T::Boolean) }
     def should_bundle_update?
-      # If both `ruby-lsp` and `debug` are in the Gemfile, then we shouldn't try to upgrade them or else it will produce
-      # version control changes
-      return false if @dependencies["ruby-lsp"] && @dependencies["debug"]
+      # If `ruby-lsp`, `ruby-lsp-rails` and `debug` are in the Gemfile, then we shouldn't try to upgrade them or else it
+      # will produce version control changes
+      return false if @dependencies.values_at("ruby-lsp", "ruby-lsp-rails", "debug").all?
 
-      # If the custom lockfile doesn't include either the `ruby-lsp` or `debug`, we need to run bundle install before
-      # updating
-      return false if custom_bundle_dependencies["ruby-lsp"].nil? || custom_bundle_dependencies["debug"].nil?
+      # If the custom lockfile doesn't include `ruby-lsp`, `ruby-lsp-rails` or `debug`, we need to run bundle install
+      # before updating
+      return false if custom_bundle_dependencies.values_at("ruby-lsp", "ruby-lsp-rails", "debug").all?(&:nil?)
```

Comment:
For this one, we want to check if any of the custom bundle dependencies are missing. This change is checking if all of them are missing, which would make us not run bundle install if only one of them is missing.
```suggestion
      return false if custom_bundle_dependencies["ruby-lsp"].nil? || custom_bundle_dependencies["debug"].nil? || (@dependencies["rails"] && custom_bundle_dependencies["ruby-lsp-rails"].nil?)
```

Reply:
For this one, I think's actually clear to branch on rails/non-rails, even though there's a bit of duplication (updated pushed).